adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -583,9 +655,22 @@ private boolean isBrokerWithPRTSupport(final Intent intent) {
      */
     public String getCurrentUser() {
         // authenticator is not used if there is not any user
-        Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
-        if (accountList.length > 0) {
-            return accountList[0].name;
+        if (isBrokerAccountServiceSupported()) {
```

Comment:
You are getting back a special error code(DEVELOPER_BROKER_PERMISSIONS_MISSING) when permisison is missing, you can request permisison based on the error code. No need to expose a new public API. 

---
In reply to: [101668304](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/734#discussion_r101668304) [](ancestors = 101668304)